### PR TITLE
Fix primefaces #4875 InputNumber: Doesn't allow minus sign in currency mode, fixed typos and ability to remove - sign

### LINKED ISSUE #4875: InputNumber: Doesn't allow minus sign in currency mode
### Describe the bug

Hello.
I faced problem in InputNumber when I use this component like this:
`<InputNumber
        mode="currency"
        currency="USD"
/>`
It uses `prefix` in input under the hood. So the problem is related not only to the `currency` mode but also when I use any prefix.

When I populate some value to the field and after that want to make this value negative by adding `-` minus sign - it happens nothing. After some investigation I found out that problem might be related to this part of code:
image
because `selectionStart=1`, `prefix` has a position 0 and minus sign should be on -1 position.

Another problem, when I have a negative value with prefix inside an input, e.g. `-$123` I cant remove value from input by pressing `backspace`. I need highlight the whole value and after that press `backspace` or `delete` button.

### Reproducer

https://codesandbox.io/s/primereact-demo-forked-pj7tdw?file=/src/App.js

### PrimeReact version

9.6.2

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

_No response_

### Steps to reproduce the behavior

1. Type some number
2. Set cursor at the start of number(before the number)
3. Try press minus sign

### Expected behavior

It should be possible to add minus sign when cursor is at the start(before the number).
It shuold be possible to remove negative number by pressing `backspace`.

### PR DESCRIPTION
**Pull request by akshayaqburst**
### Defect Fixes
Fix primefaces #4875 Fix primefaces #4875 InputNumber: Doesn't allow minus sign in currency mode.
fix #4875 

-Fixed typo issue with previous pr
-Added ability to remove minus sign in currency mode

### CODE REVIEW

**Comment by akshayaqburst** [comment]
@melloware I have fixed the typo issue with the previous PR and also added the ability to remove - sign as requested by @VasiliPodoplielov.
Thanks,

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/inputnumber/InputNumber.js [hunk 1]
```diff
@@ -402,6 +402,8 @@ export const InputNumber = React.memo(
                             } else {
                                 newValueStr = inputValue.slice(0, selectionStart - 1) + inputValue.slice(selectionStart);
                             }
+                        } else if (props.mode === 'currency' && ['¥', '$', 'I', 'N', 'R'].includes(deleteChar)) {
```
[the remark this answers is not in the document]
Comment (akshayaqburst) [reply components/lib/inputnumber/InputNumber.js]: Done @melloware

**Comment by melloware** [comment]
OK there are bigger issues here I need to investigate.

**Comment by melloware** [comment]
Superseded by this PR: https://github.com/primefaces/primereact/pull/4904

**Pull request closed without merging**